data_structures/bloom_filters/bloom_filter/main.c

```diff
@@ -0,0 +1,51 @@
+/**
+ * @file main.c
+ * @brief Test and example code for bloom filter
+ * [bloom filter](https://en.wikipedia.org/wiki/Bloom_filter)
+ * @details A Bloom filter is a space-efficient probabilistic data structure,
+ * that is used to test whether an element is a member of a set. False positive
+ * matches are possible, but false negatives are not – in other words, a query
+ * returns either "possibly in set" or "definitely not in set". Elements can be
+ * added to the set, but not removed (description of bloom filter from
+ * wikipedia)
+ * @author [Eric Breyer](https://github.com/ericbreyer)
+ * @see bloom_filter.c, bloom_filter.h
+ */
+
+#include <assert.h>  /// assertions
+#include <stdio.h>   /// printing
+
+#include "bloom_filter.h"  /// include the structure
+
+int test()
+{
+    struct BloomFilter* bf = construct_bloomFilter(100, .001);
+    int testNum;
+    for (int i = 0; i < 10; ++i)
+    {
+        testNum = i;
+        bloom_insert(bf, &testNum, sizeof testNum);
+    }
+    bloom_printStats(bf);
+    assert(bloom_contains(bf, &testNum, sizeof testNum) == PROBABLY_PRESENT);
+    testNum = 1000;
+    assert(bloom_contains(bf, &testNum, sizeof testNum) ==
+           DEFINITELY_NOT_PRESENT);
+    delete_bloomFilter(bf);
```

Comment:
I would suggest a test() like this, it's good to print more stuff and let the user what's going on at each step, by printing the elements inserted into the filter you make it easy for them to see if your implementation is working or not. There's actually no need for ifs since the assertions forced res to have that value, but I suppose it's good for formality's sake
```suggestion
    printf("Constructing a new filter.\n");
    struct BloomFilter* bf = construct_bloomFilter(100, .001);
    int testNum;
    printf("Inserting elements into the filter:")
    for (int i = 0; i < 10; ++i)
    {
        testNum = i;
        printf("%d,",i);
        bloom_insert(bf, &testNum, sizeof testNum);
    }
    printf("\n");
    bloom_printStats(bf);
    testNum = 5;
    printf("Checking %d's membership:\n",testNum);
    enum bloomResponse res =
            bloom_contains(example, &testNum, sizeof testNum);
    assert(res == PROBABLY_PRESENT);
    if(res == PROBABLY_PRESENT) {
        printf("%d is probably present.\n",testNum);
    } else {
        printf("%d is definitely not present.\n",testNum);
    }
    testNum = 1000;
    printf("Checking %d's membership:\n",testNum);
    res = bloom_contains(example, &testNum, sizeof testNum);
    assert(res == DEFINITELY_NOT_PRESENT);
    if(res == PROBABLY_PRESENT) {
        printf("%d is probably present.\n",testNum);
    } else {
        printf("%d is definitely not present.\n",testNum);
    }
    printf("Deleting the filter.\n");
    delete_bloomFilter(bf);
    printf("Testing successfully completed!");
```

Reply:
Oh wait I stitched two pieces of code together without checking variable names:sweat_smile:, lemme fix this